pxtcompiler/emitter/emitter.ts

```diff
@@ -3039,7 +3039,7 @@ ${lbl}: .short 0xffff
             } else if (node.expression.kind == SK.ElementAccessExpression) {
                 const inner = node.expression as ElementAccessExpression
                 objExpr = inner.expression
-                keyExpr = () => emitExpr(inner.argumentExpression)
+                keyExpr = () => emitExpr(ts.isStringLiteral(inner.argumentExpression) || ts.isTemplateExpression(inner.argumentExpression) ? inner.argumentExpression : ts.createTemplateExpression(ts.createTemplateHead(""), [ts.createTemplateSpan(inner.argumentExpression, ts.createTemplateTail(""))]))
```

Comment:
Would this work?

```ts
                keyExpr = () => emitExpr(asString(inner.argumentExpression))
```

Reply:
I tried, as it was actually the first commit, but it didn't work on micro:bit.

Reply:
@mmoskal ping